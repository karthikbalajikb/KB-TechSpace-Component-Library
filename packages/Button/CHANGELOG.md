# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.2.0 (2019-09-08)

### Bug Fixes

- update primary color for button ([03a748a](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/commit/03a748a))

### Features

- **Button:** add button component to packages ([5fadcc9](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/commit/5fadcc9))

## [1.1.1](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/compare/Button@1.1.0...Button@1.1.1) (2019-09-08)

### Bug Fixes

- update primary color for button ([03a748a](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/commit/03a748a))
