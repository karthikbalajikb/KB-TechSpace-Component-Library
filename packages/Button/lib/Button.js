var __makeTemplateObject =
  (this && this.__makeTemplateObject) ||
  function(cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', { value: raw });
    } else {
      cooked.raw = raw;
    }
    return cooked;
  };
import React from 'react';
import styled from 'styled-components';
import Text from '@kb-techspace-org/text';
var Button = function(_a) {
  var label = _a.label;
  return React.createElement(
    Container,
    null,
    React.createElement(Text, null, label),
  );
};
export default Button;
var Container = styled.button(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  background-color: rgb(255, 51, 101);\n  color: #fff;\n  border-radius: 4px;\n  cursor: pointer;\n  font-size: 1rem;\n  font-weight: 300;\n  padding: 9px 36px;\n',
      ],
      [
        '\n  background-color: rgb(255, 51, 101);\n  color: #fff;\n  border-radius: 4px;\n  cursor: pointer;\n  font-size: 1rem;\n  font-weight: 300;\n  padding: 9px 36px;\n',
      ],
    )),
);
var templateObject_1;
//# sourceMappingURL=Button.js.map
