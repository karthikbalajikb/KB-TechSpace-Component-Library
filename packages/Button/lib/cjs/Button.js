'use strict';
var __makeTemplateObject =
  (this && this.__makeTemplateObject) ||
  function(cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', { value: raw });
    } else {
      cooked.raw = raw;
    }
    return cooked;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
var react_1 = __importDefault(require('react'));
var styled_components_1 = __importDefault(require('styled-components'));
var text_1 = __importDefault(require('@kb-techspace-org/text'));
var Button = function(_a) {
  var label = _a.label;
  return react_1.default.createElement(
    Container,
    null,
    react_1.default.createElement(text_1.default, null, label),
  );
};
exports.default = Button;
var Container = styled_components_1.default.button(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  background-color: rgb(255, 51, 101);\n  color: #fff;\n  border-radius: 4px;\n  cursor: pointer;\n  font-size: 1rem;\n  font-weight: 300;\n  padding: 9px 36px;\n',
      ],
      [
        '\n  background-color: rgb(255, 51, 101);\n  color: #fff;\n  border-radius: 4px;\n  cursor: pointer;\n  font-size: 1rem;\n  font-weight: 300;\n  padding: 9px 36px;\n',
      ],
    )),
);
var templateObject_1;
//# sourceMappingURL=Button.js.map
