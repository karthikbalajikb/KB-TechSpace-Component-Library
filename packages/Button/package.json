{
  "name": "@kb-techspace-org/button",
  "version": "1.6.1",
  "main": "lib/index.js",
  "module": "src/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "start": "tsc -w"
  },
  "dependencies": {
    "@kb-techspace-org/text": "^1.1.1",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "styled-components": "^4.3.2"
  },
  "peerDependencies": {
    "react": "^15.0.0 || ^16.0.0",
    "react-dom": "^15.0.0 || ^16.0.0",
    "styled-components": "^4.0.0"
  },
  "gitHead": "c06eb598aef73d19bbc4528355e226a9e3c22efc"
}
