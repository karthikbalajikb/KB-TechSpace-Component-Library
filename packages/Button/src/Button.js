import styled from "styled-components";

const Button = styled.button`
  background-color: rgb(255, 51, 101);
  color: #fff;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 300;
  padding: 9px 36px;
`;

export default Button;
