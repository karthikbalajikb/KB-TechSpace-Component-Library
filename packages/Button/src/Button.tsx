import React from 'react';
import styled from 'styled-components';
import Text from '@kb-techspace-org/text';

interface Props {
  label: string;
}

const Button: React.FC<Props> = ({ label }) => {
  return (
    <Container>
      <Text>{label}</Text>
    </Container>
  );
};

export default Button;

const Container = styled.button`
  background-color: rgb(255, 51, 101);
  color: #fff;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 300;
  padding: 9px 36px;
`;
