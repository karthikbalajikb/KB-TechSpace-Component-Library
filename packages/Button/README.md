# `Button`

> Primary button component

## Usage

```js
import React, { Component } from 'react';
import Button from '@kb-techspace-org/button';

export default class App extends Component {
  render() {
    return <Button label="Primary Button" />;
  }
}
```
