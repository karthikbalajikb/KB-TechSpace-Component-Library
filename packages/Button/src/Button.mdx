import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import Button from '.';

<Meta title="MDX|Checkbox" component={Button} />

# Button

With `MDX` we can define a story for `Button` right in the middle of our
markdown documentation.

<Preview>
  <Story name="all checkboxes">
    <form>
      <Button label="demo" />
    </form>
  </Story>
</Preview>
