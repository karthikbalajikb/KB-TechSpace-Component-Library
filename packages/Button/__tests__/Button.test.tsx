import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from '../src/Button';

describe('Button', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Button label="demo" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
