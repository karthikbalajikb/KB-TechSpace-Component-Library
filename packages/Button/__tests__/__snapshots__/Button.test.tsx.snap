// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c1 {
  font-size: 14px;
  line-height: 27px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
  color: lightgray;
  opacity: 1;
}

.c0 {
  background-color: rgb(255,51,101);
  color: #fff;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1rem;
  font-weight: 300;
  padding: 9px 36px;
}

<button
  className="c0"
>
  <span
    className="c1"
    color="lightgray"
    fontSize={14}
    fontWeight={400}
    letterSpacing={0.3}
    opacity={1}
  >
    demo
  </span>
</button>
`;
